src/diffusers/pipelines/stable_diffusion_3/pipeline_stable_diffusion_3.py

```diff
@@ -911,13 +972,30 @@ def __call__(
                 # broadcast to batch dimension in a way that's compatible with ONNX/Core ML
                 timestep = t.expand(latent_model_input.shape[0])
 
+                if ip_adapter_image is not None or ip_adapter_image_embeds is not None:
+                    ip_hidden_states, temb = self.transformer.image_proj(
+                        image_embeds,
+                        timestep.to(dtype=latents.dtype),
+                        need_temb=True,
+                    )
+
+                    image_prompt_embeds = dict(
+                        ip_hidden_states = ip_hidden_states,
+                        temb = temb
+                    )
+                else:
+                    image_prompt_embeds = {}
```

Comment:
If `self.joint_attention_kwargs` is not None we can `update` it, otherwise we can set the value to `image_prompt_embeds`, then we don't need the `else` path.

Reply:
Note we will remove the `None` checks in the future if we end up changing the default